java-manta-client/pom.xml

```diff
@@ -170,6 +170,20 @@
             <version>${dependency.urlbuilder.version}</version>
         </dependency>
 
+        <!-- object cloning library for memoization of encryption state -->
+        <dependency>
+            <groupId>uk.com.robust-it</groupId>
```

Comment:
I couldn't find any other libraries for deep-cloning besides this one and the copy functionality provided by kryo. Utilizing kryo would effectively mean bringing `java-manta-client-kryo-serialization` into the main module.

I've limited usage of this library to only cloning `Cipher` objects produced by the BouncyCastle library. The `HmacCloner` and `DigestCloner` classes in this PR leverage the `Memoable` interface and some reflection to clone `Hmac` objects instead.